### ISSUE TITLE
Messages not silenced when interrogating source caches

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#280]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/280)
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Mar 4, 2018, 17:16

This is a regression from centralizing state handling in the `Element` class.

The following status messages are extra verbose and intended to be silenced by default:

```
[--:--:--][        ][ main:core-deps/gstreamer.bst-0     ] STATUS  Running host command

    /usr/bin/git cat-file -t a1bf0f0e6a8118eab309707f52a64e1ce2c2be95

[--:--:--][        ][ main:core-deps/gstreamer.bst-0     ] STATUS  Running host command

    /usr/bin/git show a1bf0f0e6a8118eab309707f52a64e1ce2c2be95:.gitmodules

[--:--:--][        ][ main:core-deps/gstreamer.bst-0     ] STATUS  Running host command

    /usr/bin/git ls-tree a1bf0f0e6a8118eab309707f52a64e1ce2c2be95 common
```

Since we centralized these source cache state interrogations which we run in the main process, the messages are no longer silenced by default.

We probably need a context manager for silencing the less important messages (as we have an option for in the `timed_activity` context managers), so we can do this generically from `element.py` when updating the state of an element in the main process.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Mar 4, 2018, 17:19

Note: I already worked around this at initial load time with a `timed_activity()` context manager, these messages are still happening as state gets updated while the session proceeds.

This will notably happen a lot, adding a lot of obnoxious noise to the main log when running `bst build --track ...`

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Mar 7, 2018, 11:58

closed via commit 7fdddf30992efb9f2a88686638f997208b01d0f2

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @knownexus]](https://gitlab.com/knownexus) on Mar 7, 2018, 15:23

mentioned in commit 7fdddf30992efb9f2a88686638f997208b01d0f2
